src/science/misc/cable_climate.F90

```diff
@@ -15,271 +15,475 @@
-!$PPc    = Gaero / ( Gaero + 4.0*SBoltz*((TempA+273.16)**3)/(RhoA*Capp) )
-!$                                                    ! PPc = Ga/(Ga+Gr)      [-]
-!$EpsA   = Epsif(TempA, Pmb)                          ! Epsi at TempA         [-]
-!$PhiSd  = SolarMJ * 1.0e6 / (DayltFrac*SecDay)       ! daylt down solar      [W/m2]
-!$PhiLd  = 335.97 * (((TempA + 273.16) / 293.0)**6)   ! daylt down thermal    [W/m2]
-!$                                                    !   (Swinbank formula)
-!$PhiAi  = (1.0-Albedo)*PhiSd + Emis *    &           ! daylt iso-avail engy  [W/m2]
-!$         (PhiLd - SBoltz*((TempA + 273.16)**4))     !   (veg + soil)
-!$PhiEq  = PhiAi * (PPc*EpsA) / (PPc*EpsA + 1.0)      ! equil ltnt heat flux  [W/m2]
-!$PhiEq  = max(PhiEq, 1.0)                            ! PhiEq > +1 W/m2, so non-negative
-!$                                    ! precipitation         [m/day]
-!$FWPT   = CoeffPT * PhiEq * ((DayltFrac*SecDay) / (RhoW*Rlat))
-
     ! accumulate annual evaporation and potential evaporation
-    !ppc = 1.0
     RhoA = met%pmb * 100.0 / (8.314 * (met%tk)) ! air density [molA/m3]
-    PPc    = Gaero / ( Gaero + 4.0*SBoltz*((met%tk**3)/(RhoA*Capp) ))
+    PPc  = Gaero / ( Gaero + 4.0*SBoltz*(met%tk**3)/(RhoA*Capp) )
 
-    EpSA = Epsif(met%tk - 273.16, met%pmb)
+    EpSA  = epsif(met%tk - 273.16, met%pmb)
     phiEq = canopy%rniso * (PPc*EpsA) / (PPc*EpsA + 1.0)      ! equil ltnt heat flux  [W/m2]
 
-    IF (idoy==1 .AND. MOD(ktau,ktauday)==1 ) THEN
-       !  climate%evap_PT =  max(phiEq,1.0)*CoeffPT/air%rlam*dels  ! mm
-       climate%evap_PT =  phiEq*CoeffPT/air%rlam*dels  ! mm
-       !  climate%evap_PT = canopy%epot  ! mm
-       !  climate%aevap  =   canopy%fe/air%rlam*dels ! mm
-       climate%aevap = met%precip ! mm
-    ELSE
-
-       climate%evap_PT = climate%evap_PT + MAX(phiEq,1.0)*CoeffPT/air%rlam*dels  ! mm
+    if ((idoy == 1) .and. (mod(ktau, ktauday) == 1)) then
+       ! first time step of year
+       climate%evap_PT       = phiEq * CoeffPT / 2.5014e6 * dels  ! mm
+       climate%aevap         = met%precip ! mm
+       climate%fapar_ann_max = 0.0
+    else
+       climate%evap_PT = climate%evap_PT + phiEq * CoeffPT / 2.5014e6 * dels  ! mm
+       ! climate%evap_PT = climate%evap_PT + max(phiEq,1.0)*CoeffPT/air%rlam*dels  ! mm
        ! climate%evap_PT =climate%evap_PT + canopy%epot  ! mm
        ! climate%aevap =  climate%aevap + canopy%fe/air%rlam*dels ! mm
        climate%aevap = climate%aevap + met%precip ! mm
-    ENDIF
+    endif
 
     ! accumulate daily temperature, evap and potential evap
-    IF(MOD(ktau,ktauday)==1) THEN
-       climate%dtemp = met%tk - 273.15
-       climate%dmoist = canopy%fwsoil
-    ELSE
-       climate%dtemp = climate%dtemp + met%tk - 273.15
-       climate%dmoist = climate%dmoist + canopy%fwsoil
-    ENDIF
+    if (mod(ktau, ktauday) == 1) then
+       climate%dtemp     = met%tk - 273.15
+       climate%dmoist    = sum(real(ssnow%wb(:,:))*veg%froot(:,:), 2)
```

Comment:
I think we need to retain a comment here around the change in science from `%dmoist = %fwsoil` to `%dmoist  = SUM(%wb * %froot, 2)`.  These quantities are related but different.

Reply:
such as?

OK before we get too involved, I am. totally unfamiliar with this module - who uses cable_climate?

Reply:
As far as I know only CABLE-POP simulations actively use this routine - so that's the TRENDY and BIOS applications - however the `climate%` is woven throughout the offline code.

I'm not actually sure how CM2 and AM3 are handling this to be honest - the TYPE is there (if incomplete) and there is an allocation/initialisation.  However there is no call to the `cable_climate()` subroutine yet there is a use (in AM3) in `casa_rplant` - I would have thought that would have caused some 'interesting' results or a crash.

Reply:
To answer the implicit question - all we need to do is include a comment such as
`! Has been updated from %dmoist  = canopy%fwsoil `  with, if possible, a svn revision number.

Reply:
> there is a use (in AM3) in casa_rplant and `cbl_dryLeaf`

Specifically `casa_rplant()` and `cbl_dryLeaf` use `climate%qtemp_max_last_year` - this is initialised to 0.0.  HOwever both uses are inside a IF `cable_user%CALL_climate` condition so will only be active if triggered by the user.